### PR TITLE
Improving c2f.py

### PR DESCRIPTION
**Pull request by hxu-ids**

### CODE REVIEW

File: ox_ui/core/c2f.py [hunk 1]
```diff
@@ -192,20 +194,27 @@ def __init__(self, clickCmd, skip_opt_re=None, tweaks: list = None,
...
-    def form_cls(self):
-
-        class ClickForm(FlaskForm):
-            """Form to run Click command.
-            """
-
+    def click_cmd_params(self):
+        """
+        to make sure skip_opt_re and gobbled specs are respected
+        """
```
Comment (hxu-ids) [inline ox_ui/core/c2f.py:200]: to be called in multiple places to make sure `skip_opt_re` and `gobbled` logic respected

File: ox_ui/core/c2f.py [hunk 2]
```diff
@@ -254,42 +263,43 @@ def form(self):
         cls = self.form_cls()
         return cls()
 
-    def click_opt_to_wtf_field(self, opt):
+    @staticmethod
+    def click_opt_to_wtf_field(opt):
+        validators = []
+        if opt.required:
+            validators.append(
+                DataRequired(f'* required field'))
```
Comment (hxu-ids) [inline ox_ui/core/c2f.py:271]: this will make fields in GUI `required`, browser will prompt user to enter values

File: ox_ui/core/c2f.py [hunk 3]
```diff
@@ -254,42 +263,43 @@ def form(self):
...
         if opt.type == types.INT:
-            field = IntegerField(opt.name, validators=[], description=str(
-                opt.help), default=opt.default)
-        elif opt.type == types.STRING:
-            field = StringField(opt.name, validators=[], description=str(
-                opt.help), default=opt.default)
-        elif isinstance(opt.type, types.DateTime) or (
+            return IntegerField(opt.name, **kwargs)
+        if opt.type == types.STRING:
+            field_cls = PasswordField if opt.hide_input else StringField
```
Comment (hxu-ids) [inline ox_ui/core/c2f.py:280]: added support to `PasswordField`

**Review by emin63 — APPROVED** [review]
Nice. LGTM. Approving and will merge and push to PYPI.

**Comment by emin63** [comment]
@hxu-ids : I was not able to push this out since tests are failing.

I setup GitHub actions to auto-run tests. Can you fix the failing test? My guess is that the `--also` arg in `tests/flask/c2f_app.py` is not being handled correctly since `multiple` options are a bit tricky.

**Pull request merged**